### PR TITLE
Allow custom caret visibility dependencies and CSS for caret blink rate

### PR DESCRIPTION
**Pull request by JordanMartinez**

### CODE REVIEW

File: richtextfx/src/main/java/org/fxmisc/richtext/CssProperties.java [hunk 1]
```diff
@@ -109,4 +109,33 @@ public String getName() {
...
+        }
+
+        @Override
+        public Object getBean() {
+            return bean;
+        }
+
+        @Override
+        public String getName() {
+            return "highlightTextFill";
```
Comment (TomasMikula) [inline richtextfx/src/main/java/org/fxmisc/richtext/CssProperties.java]: copy&paste typo? :D

Comment (JordanMartinez) [inline richtextfx/src/main/java/org/fxmisc/richtext/CssProperties.java]: Maybe! :sweat_smile:

File: richtextfx/src/main/java/org/fxmisc/richtext/CssProperties.java [hunk 2]
```diff
@@ -109,4 +109,33 @@ public String getName() {
...
+    static class BlinkRateProperty extends StyleableObjectProperty<javafx.util.Duration> {
+        private final Object bean;
+
+        private final CssMetaData<? extends Styleable, javafx.util.Duration> cssMetaData;
+
+        public BlinkRateProperty(Object bean, javafx.util.Duration initialValue) {
+            super(initialValue);
+            this.bean = bean;
+            cssMetaData = new PropertyCssMetaData<>(
+                    this, "-fx-blink-rate",
```
Comment (JordanMartinez) [inline richtextfx/src/main/java/org/fxmisc/richtext/CssProperties.java]: What's your opinion? Should "blink-rate" be renamed to "caret-blink-rate" to better specify which object's blink rate is being customized?

Comment (JordanMartinez) [inline richtextfx/src/main/java/org/fxmisc/richtext/CssProperties.java]: Meh... I just went and changed the name. Seems like a better overall decision.

**Comment by TomasMikula** [comment]
👍

**Pull request merged**